background-script.js

```diff
@@ -11,23 +11,82 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-async function getCurrentWindow()
-{
-    let windows = await messenger.windows.getAll();
+import {parse5322} from "./email-address-parser/email-addresses.mjs";
 
-    for (let window of windows) {
-        if ((window.type === "messageDisplay" || window.type === "normal")
-            && window.focused === true)
-            return window;
+// There is no real need for these functions to be async. But being able to use
+// the .then() callback of the return value for further manipulations is quite handy.
+async function getFirstHeader(arr) {
+    if (Array.isArray(arr) && arr.length > 0) {
+        return arr[0];
+    }
+    return undefined;
+}
+async function getAllHeader(arr) {
+    if (Array.isArray(arr) && arr.length > 0) {
+        return arr;
+    }
+    return undefined;
+}
+function parseDisplayName(addr) {
+    let rv = parse5322.parseOneAddress(addr);
+    return {
+        name: rv.name,
+        email: rv.address,
+    }
+}
+// Find first text/plain body.
+function getBody(parts) {
+    // First check all parts in this level.
+    for (let part of parts) {
+        if (part.body && part.contentType == "text/plain") {
+            return part.body;
+        }
+    }
+    // Now check all subparts.
+    for (let part of parts) {
+        if (part.parts) {
+            let body = getBody(part.parts);
+            if (body) {
+                return body
+            }
+        }
     }
     return null;
 }
+// Compatibility layer for former experiment.
```

Comment:
Reads like as if there could be a better way. However, to my understanding, there is no way for the content script to access `browser.messages` directly. Therefore, we need such an interface, no?